### ISSUE TITLE
PLUGIN_INITIALIZATION_ERROR when trying to start server

### ISSUE THREAD

**Issue by danikk741** [deepstreamIO/deepstream.io#981]
I have installed deepstream server v 4.0.6 and mongodb connector. Added following code to config.yml: 
`storage:`
`  name: mongodb`
`  options:`
`    connectionString: 'mongodb://localhost'`
`    database: 'deep'`
`    defaultTable: 'deep'`
`    splitChar: '/'`

And when I try ro start server I keep getting message: 
> PLUGIN_INITIALIZATION_ERROR | storage needs to implement async whenReady and close, please look at the DeepstreamPlugin API here

I read docs, other issues, stackoverflow questions and did not understand what I should do. Can someone help me please?

**Comment by ismael-iskauskas** [comment]
Hello,
try this:
`connectionString: 'mongodb://localhost/deep'`
worked for me

**Comment by danikk741** [comment]
Thanks for answer but it did not help. Keep getting this message

**Comment by ismael-iskauskas** [comment]
Which version are you using? In your package.json make sure its something like

`"@deepstream/storage-mongodb": "^2.0.2"`

in config.yml

```
storage:
  name: mongodb
  options:
    connectionString: mongodb://127.0.0.1:27017/deepstream
    database: 'deepstream'
    splitChar: '/'
```

you are missing the port in connectionString

**Comment by danikk741** [comment]
I had v 1, because I nstalled it using `deepstream install storage mongodb`. Now I rebuild everything using node and it looks like it works but then i get next error:
`D:\deep\node_modules\@deepstream\storage-mongodb\src\connector.js:67`
`this.on('error', (error) => this.services.logger.fatal(error))`
`TypeError: Cannot read property 'logger' of undefined`

I tried to fix it but it is still there.

**Comment by ismael-iskauskas** [comment]
can you post your config.yml?

**Comment by danikk741** [comment]
```
serverName: UUID
showLogo: true
logLevel: INFO
dependencyInitialisationTimeout: 2000
libDir: ../lib

rpc:
  ackTimeout: 1000
  responseTimeout: 10000

record:
  cacheRetrievalTimeout: 30000
  storageRetrievalTimeout: 30000

listen:
  shuffleProviders: true
  responseTimeout: 500
  rematchInterval: 60000
  matchCooldown: 10000

connectionEndpoints:
  - type: ws-websocket
    options:
        port: 6020
        host: 0.0.0.0
        urlPath: /deepstream
        healthCheckPath: /health-check
        heartbeatInterval: 30000
        outgoingBufferTimeout: 50
        maxBufferByteSize: 100000

        headers:
          - user-agent

        unauthenticatedClientTimeout: 180000
        maxAuthAttempts: 3
        logInvalidAuthData: false
        maxMessageSize: 1048576

  - type: node-http
    options:
      port: 8080
      host: 0.0.0.0
      allowAuthData: true
      enableAuthEndpoint: false
      authPath: /auth
      postPath: /
      getPath: /
      healthCheckPath: /health-check

      headers:
        - user-agent

      allowAllOrigins: true
      origins:
        - 'https://example.com'
      maxMessageSize: 1024

logger:
  type: default
  options:
    colors: true
    logLevel: INFO

storage:
  name: mongodb
  options:
    connectionString: 'mongodb://127.0.0.1:27017/deep'
    database: 'deep'
    splitChar: '/'

auth:
  type: none

permission:
  type: config
  options:
    path: ./permissions.yml
    maxRuleIterations: 3
    cacheEvacuationInterval: 60000
```

**Comment by ismael-iskauskas** [comment]
What i did to get it up and running. I created a server.js

```
const { Deepstream } = require('@deepstream/server');
const path = require('path');

const config = path.resolve('.', 'config/config.yml');
const server = new Deepstream(config);

// start the server
server.start()
```
Folder structure is like this:
config
 -- config.yml
 -- permission.yml
 -- users.yml
server.js

run with `node server.js`. it worked for me

**Comment by yasserf** [comment]
I think the issue is around this:

```
deepstream install storage mongodb
```

I don't think we published all the 4.0 artefacts correctly, it's alot of work and very few people use it. For now could you please do what @ismael-iskauskas recommended? I'll try to solve this in the near future to work with the binary correctly.

You can also do an npm install @deepstream/storage-mongo --production, zip the directory and move into the lib directory if you want to use the binary directly.

**Comment by danikk741** [comment]
Thank you! It works now
